R/moma_arguments.R

```diff
@@ -1,80 +1,83 @@
+#' @name moma_sparsity_options
+#' @param ... Force users to specify arguments by names.
+#' @param lambda A vector containing penalty values
+#' @param select_scheme A char being either "b" (nested BIC search) or "g" (grid search).
+#'
+#' Selection of tuning parameters (four parameters in our case, \code{alpha_u}, \code{alpha_v}, \code{lambda_u}, \code{lambda_v}) is a problem in MoMA. Currently we provide
+#' two modes: grid search ("g") and nested BIC search ("b"). Grid search means we solve the problem
+#' for all combinations of parameter values provided by the user.
+#'
+#' To explain nested BIC search, we need to look into how the algorithm runs.
+#' To find an (approximate) solution to a penalized SVD (Singular Value Decomposition) problem is to solve two
+#' penalized regression problems iteratively. Let's call them problem u and problem v, which give
+#' improving estimates of the right singular vector, \emph{u}, and the left singular vector, \emph{v}, respectively.
+#' For each regression problem, we can select the optimal parameters
+#' based on BIC.
+#'
+#' The nested BIC search is essentially two 2-D searches. We start from SVD solutions, and then find the optimal
+#' parameters for problem u, given current estimate of \emph{v}. Using the result from previous step, update
+#' current estimate of \emph{u}, and then do the same thing for problem v,
+#' that is, to find the optimal parameters for problem v given current estimate of \emph{u}. Repeat
+#' the above until convergence or the maximal number of iterations has been reached.
+NULL
+
+#' Introduction to selection schemes in MoMA
+#'
+#' Please see the description of the argument \code{select_scheme} in
+#' \code{\link{moma_sparsity_options}}. The \code{select_scheme} argument
+#' presents in functions listed in \code{\link{moma_sparsity_options}}, and the function
+#' \code{\link{moma_smoothness}}.
+#' @name select_scheme
 NULL
 
 empty <- function() {
     arglist <- list()
-    class(arglist) <- "moma_sparsity"
+    class(arglist) <- "_moma_sparsity_type"
     return(arglist)
 }
 
-#' LASSO
-#'
-#' Use this function to set the penalty function as lasso
-#' \deqn{\lambda \sum \| x_{i} \| ,}
-#' where \eqn{\lambda} is set by \code{lambda_u/v} in the function \code{moma_svd}.
-#'
-#' @param ... Forces users to specify all arguments by name.
-#' @param non_negative a boolean value. Set \code{TRUE} to add non-negativity
-#' constraint.
-#'
-#' @return a \code{moma_sparsity} object, which is a list containing \code{non_negative}
-#'
-#' @examples
-#' lasso(non_negative = FALSE)
-#' @export
 lasso <- function(..., non_negative = FALSE) {
     if (length(list(...)) != 0) {
         moma_error("Please specify the correct argument by name.")
     }
 
     if (!is_logical_scalar(non_negative)) {
-        moma_error(sQuote("non_negative"), " should be a boolean value.")
+        moma_error(sQuote("non_negative"), " should be a Boolean value.")
```

Comment:
Can we use the `check_*` pattern here and in other penalties?

Reply:
Fixed.